src/ngMock/angular-mocks.js

```diff
@@ -1341,13 +1341,13 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
       // service the ngMockE2E version can pass through to the real backend
       originalHttpBackend = $delegate.$$originalHttpBackend || $delegate;
 
-  function createResponse(status, data, headers, statusText) {
+  function createResponse(status, data, headers, statusText, xhrStatus) {
     if (angular.isFunction(status)) return status;
 
     return function() {
       return angular.isNumber(status)
-          ? [status, data, headers, statusText]
-          : [200, status, data, headers];
+          ? [status, data, headers, statusText, xhrStatus || (status === 200 ? 'success' : 'error')]
```

Comment:
@gkalpak 

> Then you should change both lines to xhrStatus || 'complete'.

Does it still make sense to use `xhrStatus || 'complete'` ?

```
function createResponse(status, data, headers, statusText, xhrStatus) {
    if (angular.isFunction(status)) return status;

    return function() {
      return angular.isNumber(status)
          ? [status, data, headers, statusText, xhrStatus || 'complete']
          : [200, status, data, headers, xhrStatus || 'complete'];
    };
  }
```

As we're always returning a status code, I guess we can consider it 'complete' in any situation? Is there any need to manually set the xhrStatus to anything else but 'complete'  ?